firewood/src/merkle.rs

```diff
@@ -71,6 +72,16 @@ pub struct Merkle<S, T> {
     phantom: PhantomData<T>,
 }
 
+impl<T> Merkle<MutStore, T> {
```

Comment:
Same pattern here. If you `impl<T> From<Merkle<MutStore, T>> for Merkle<SharedStore, T>` you get into (and from, if you happen to need it) for free.